### PR TITLE
Fix match / selections bug

### PR DESCRIPTION
**Pull request by MrCreosote**
Previous change to specified fields left out the special match and selection fields

### CODE REVIEW

File: src/service/data_products/common_functions.py [hunk 1]
```diff
@@ -268,16 +269,20 @@ async def query_simple_collection_list(
...
-                        names.FLD_MATCHES_SELECTIONS]
+                    d[selection_field] = _get_matchsel(filters.selection_spec, d)
             acceptor(d)
     finally:
         await cur.close(ignore_missing=True)
 
 
+def _get_matchsel(spec: SubsetSpecification, doc: dict[str, Any]):
+    if not doc.get(names.FLD_MATCHES_SELECTIONS):
+        return False
```
Comment (Tianhao-Gu) [inline src/service/data_products/common_functions.py:282]: not `list()` but `False` here?

File: src/service/data_products/common_functions.py [hunk 2]
```diff
@@ -352,6 +357,10 @@ async def query_table(
         raise errors.IllegalParameterError(
                 f"No such field for collection {filters.collection_id} load version "
                 + f"{filters.load_ver}: {filters.sort_on}")
+    if not filters.selection_spec.is_null_subset():
+        fields = [names.FLD_SELECTED_SAFE] + fields
```
Comment (Tianhao-Gu) [inline src/service/data_products/common_functions.py:361]: it's ok to have duplicate `names.FLD_SELECTED_SAFE`/ `names.FLD_MATCHED_SAFE` in `fields` array?

Comment (MrCreosote) [reply src/service/data_products/common_functions.py:361]: I'm not understanding, those aren't duplicate names

File: src/service/data_products/common_functions.py (hunk 1 above)
Comment (MrCreosote) [reply src/service/data_products/common_functions.py:282]: No, `__match__` and `__sel__` field values are expected to be `True` or `False`

File: src/service/data_products/common_functions.py (hunk 2 above)
Comment (Tianhao-Gu) [reply src/service/data_products/common_functions.py:361]: can `fields` array already have `names.FLD_SELECTED_SAFE`? unlikely?

File: src/service/data_products/common_functions.py (hunk 1 above)
Comment (MrCreosote) [reply src/service/data_products/common_functions.py:282]: I added the missing return type to make this clearer

File: src/service/data_products/common_functions.py (hunk 2 above)
Comment (MrCreosote) [reply src/service/data_products/common_functions.py:361]: They don't actually exist in the database, so unless the calling code does something really weird, no.

Comment (MrCreosote) [reply src/service/data_products/common_functions.py:361]: Currently the field names are coming from the column specs, which don't have those names.

File: src/service/data_products/common_functions.py (hunk 1 above)
Comment (Tianhao-Gu) [reply src/service/data_products/common_functions.py:282]: 👍

File: src/service/data_products/common_functions.py (hunk 2 above)
Comment (Tianhao-Gu) [reply src/service/data_products/common_functions.py:361]: 👍

**Review by Tianhao-Gu — APPROVED** [review]
👍

**Pull request merged**